spec.bs

```diff
@@ -2660,6 +2696,38 @@ CORP violation report=] algorithm, as leaving it unfenced may cause a privacy le
   [=automatic beacon data/once=] set to true.
 </div>
 
+<div algorithm=get-initiator-ancestor-root-origin>
+  To <dfn>get the initiator ancestor root origin</dfn> of a given {{Document}} |sourceDocument|:
+
+  1. Let |navigable| be |sourceDocument|'s [=node navigable=].
+
+  1. While |navigable| is not null and |navigable| is not a [=fenced navigable container/fenced
+     navigable=]:
+  
+     1. Set |navigable| to |navigable|'s [=navigable/parent=].
+
+  1. If |navigable| is null, return null.
+
+  1. Otherwise:
+
+     1. [=Assert=]: |navigable| is a [=fenced navigable container/fenced navigable=].
+
+     1. Set |navigable| to |navigable|'s [=navigable/unfenced parent=].
+
+     1. While |navigable| is not null and |navigable| is not a [=fenced navigable container/fenced
+        navigable=]:
+
+        1. Set |navigable| to |navigable|'s [=navigable/parent=].
+
+     1. If |navigable| is null, return null.
+
+     1. Otherwise, return |navigable|'s [=navigable/active document=]'s [=Document/origin=].
+
+  Note: This algorithm is used to get the [=origin=] of the [=fenced navigable container/fenced
+  navigable=] whose descendant [=fenced navigable container/fenced navigable=]'s descendant is
+  |sourceDocument|, if one exists.
```

Comment:
I think this note is kind of confusing, sorry. Instead of targeting from top->bottom (i.e., "descendant's descendant's [...]) can you describe what this is supposed to return given a fenced frame? I.e., center the language around bottom->top. So given a fenced frame, will the return origin always come from a fenced frame ancestor? Or can it come from a main frame? When are the "If navigable is null" conditions true and how does that effect the ancestor chain. Maybe even provide an example or two like I did in the comment I left about this algorithm. It's a tricky one so I want to make sure we're as expressive as possible so we don't have to think through this again.

Reply:
Added a non-normative explanatory section with examples.